modules/net-interconnect-attachment-direct/variables.tf

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "region" {
+  description = "Region where the router resides"
+  type        = string
+  default     = "europe-west1-b"
+}
+
+variable "project_id" {
+  description = "The project containing the resources"
+  type        = string
+}
+
+variable "router_name" {
+  description = "Router name used for auto created router, or to specify an existing router to use if `router_create` is set to `true`. Leave blank to use vlan attachment name for auto created router."
+  type        = string
+  default     = "router-vlan-attachment"
+}
+
+variable "router_config" {
+  description = "Router asn and custom advertisement configuration, ip_ranges is a map of address ranges and descriptions.. "
+  type = object({
+    description = string
+    asn         = number
+    advertise_config = object({
+      groups    = list(string)
+      ip_ranges = map(string)
+      mode      = string
+    })
+  })
+
+  default = {
+    description      = null
+    asn              = 64514
+    advertise_config = null
+  }
+
+}
+
+variable "router_create" {
+  description = "Create router."
+  type        = bool
+  default     = true
+}
+
+variable "network_name" {
```

Comment:
if this is only used for the router resource, it should be named `router_network` and have a `null` default for cases where the router is managed externally. WDYT?